src/tests/ftest/control/version.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python3
+'''
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+'''
+import re
+
+from apricot import TestWithServers
+from general_utils import run_pcmd
+
+
+class DAOSVersion(TestWithServers):
+    """JIRA ID: DAOS-8539
+
+    Verify version number and the output string.
+
+    :avocado: recursive
+    """
+    def test_version(self):
+        """Verify version number for dmg, daos, daos_server, and daos_agent
+        against RPM.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=vm
+        :avocado: tags=control
+        :avocado: tags=version_number
+        """
+        errors = []
+
+        # Get RPM version.
+        rpm_command = "rpm -qa|grep daos-server"
+        output = run_pcmd(hosts=self.hostlist_servers, command=rpm_command)
+        self.log.info("RPM output = %s", output)
+        stdout = output[0]["stdout"][0]
+        self.log.info("RPM stdout = %s", stdout)
+        result = re.findall(
+            r"daos-server-[tests-|tests_openmpi-]*([\d.]+)", stdout)
+        rpm_version = result[0]
+        self.log.info("RPM version = %s", rpm_version)
+
+        # Get dmg version.
+        dmg_cmd = self.get_dmg_command()
+        output = dmg_cmd.version().stdout.decode("utf-8")
+
+        # Verify that "dmg version" is in the output.
+        if "dmg version" not in output:
+            errors.append("dmg version is not in the output! {}".format(output))
+
+        result = re.findall(r"dmg version ([\d.]+)", output)
+        dmg_version = result[0]
+        self.log.info("dmg version = %s", dmg_version)
+
+        # Get daos version.
+        daos_cmd = self.get_daos_command()
+        output = daos_cmd.version().stdout.decode("utf-8")
+
+        # Verify that "daos version" is in the output.
+        if "daos version" not in output:
+            errors.append(
+                "daos version is not in the output! {}".format(output))
+
+        result = re.findall(r"daos version ([\d.]+)", output)
+        daos_version = result[0]
+        self.log.info("daos version = %s", daos_version)
```

Comment:
Similar to `dmg version`, `result[0]` would raise an exception if not found. For this and subsequent `version` commands.

Reply:
Thanks. Fixed it.